File: SORTING/quickSort.c
#include<stdio.h> 
#include<stdlib.h>
#include<time.h>

int partition(int A[],int l, int r)
{
    int X=l, Y=r, pivot= A[l], temp;
    while (X<Y)
    {
        while (A[X]<=pivot && X<=r)
        {
            X++;
        }
        while (A[Y]>pivot && Y>l)
        {
            Y--;
        }
        if(X<Y)
        {
            temp = A[X];
            A[X]=A[Y];
            A[Y]=temp;
            return Y;
        }
    }
    temp = A[l];
    A[l]=A[Y];
    A[Y]=temp;
    return Y;

}

void QuickSort(int A[],int l,int r)
{
    int p;
    if(l<r)
    {
        p = partition(A,l,r);
        QuickSort(A,l,p-1);
        QuickSort(A,p+1,r);

    }
}

void display(int A[],int n)
{
    int i;
    for(i=0;i<n;i++)
    {
        printf("%d ",A[i]);
    }
    printf("\n");

}


int main()
{
    int A[100000],n,i;
    clock_t start,end,diff;
    printf("Enter n:"); 
    scanf("%d",&n);
    for(i=0;i<n;i++)
    {
        A[i]=rand();
    }
    display(A,n);
    start=clock();
    QuickSort(A,0,n-1);
    end=clock();
    printf("The sorted numbers are:\n");
    display(A,n);
    diff=end-start;
    printf("The time taken is %f seconds\n",(float)diff/CLOCKS_PER_SEC);
    return 0;
}

File: SORTING/SortingInPython/quickSort.py
import random
import time

def partition(A, l, r):
    X = l
    Y = r
    pivot = A[l]
    while X < Y:
        while A[X] <= pivot and X <= r:
            X += 1
        while A[Y] > pivot and Y > l:
            Y -= 1
        if X < Y:
            A[X], A[Y] = A[Y], A[X]
    A[l], A[Y] = A[Y], A[l]
    return Y

def QuickSort(A, l, r):
    if l < r:
        p = partition(A, l, r)
        QuickSort(A, l, p - 1)
        QuickSort(A, p + 1, r)

def display(A, n):
    for i in range(n):
        print(A[i])
    print()

A = [0] * 100000
print("Enter n:")
n = int(input())
for i in range(n):
    A[i] = random.randint(0, 100000)
display(A, n)
start = time.time()
QuickSort(A, 0, n - 1)
end = time.time()
print("The sorted numbers are:")
display(A, n)
print(f"The time taken is {end - start} seconds")
